Sources_v3/StreamChat/Database/DTOs/MessageDTO.swift

```diff
@@ -240,14 +241,19 @@ extension NSManagedObjectContext: MessageDatabaseSession {
             let user = try saveUser(payload: userPayload)
             dto.mentionedUsers.insert(user)
         }
+
+        try payload.threadParticipants.forEach { userPayload in
+            let user = try saveUser(payload: userPayload)
+            dto.threadParticipants.insert(user)
+        }
```

Comment:
If user had left a thread reply and then deleted it will they stay in `threadParticipants` list? 🤔 

I'd suggest to reset the `dto.threadParticipants` to ensure we are consistent with backend.

```suggestion
        dto.threadParticipants = try Set(
          payload.threadParticipants.forEach { try saveUser(payload: $0) }
        )
```